### PR TITLE
Update devise.rb

### PR DESCRIPTION
**Pull request by chambin0**
Fixes issue #505 
I fixed the email validation line in config/initializers/devise.rb

#### Describe the changes you have made in this pr -
Changed line 173 from "config.email_regexp = /\A[^@\s]+@[^@\s]+\z/" to "config.email_regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/"

### CODE REVIEW

**Comment by tachyons** [comment]
https://github.com/CircuitVerse/CircuitVerse/blob/master/app/controllers/group_members_controller.rb#L35 This is the line you need to check

**Comment by chambin0** [comment]
Okay, now i also changed the utils.rb which is actually responsible for the email validation. I think it should work correctly now.

File: app/helpers/utils.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module Utils
   # @return array of valid emails
   def self.parse_mails(mails)
     mails.split(/[\s,\,]/).select do |email|
-      email.present? && URI::MailTo::EMAIL_REGEXP.match(email)
+      email.present? && /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.match(email)
```
Comment (tachyons) [inline app/helpers/utils.rb]: You can use Devise::email_regexp instead

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -170,7 +170,7 @@
   # Email regex used to validate email formats. It simply asserts that
   # one (and only one) @ exists in the given string. This is mainly
   # to give user feedback and not to assert the e-mail validity.
-  config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
+  config.email_regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
```
Comment (tachyons) [inline config/initializers/devise.rb]: remove this change

Comment (chambin0) [reply config/initializers/devise.rb]: Why? it's a correct regular expression for email validation from the link given in the task description.

File: app/helpers/utils.rb (hunk 1 above)
Comment (chambin0) [reply app/helpers/utils.rb]: I did it but the bot says i shouldn't use "::".

File: app/helpers/utils.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ module Utils
   # @return array of valid emails
   def self.parse_mails(mails)
     mails.split(/[\s,\,]/).select do |email|
-      email.present? && URI::MailTo::EMAIL_REGEXP.match(email)
+      email.present? && Devise::email_regexp.match(email)
```
[the remark this answers is not in the document]
Comment (tachyons) [reply app/helpers/utils.rb]: Use . instead of :: and match? instead of match

**Comment by tachyons** [comment]
Update PR title, squash commits and provide meaningful commit message

**Pull request closed without merging**